### ISSUE TITLE
Deferred 'app' creation makes component setup hard

### ISSUE THREAD

**Issue by leonerd** [matrix-org/matrix-appservice-bridge#35]
TL;DR: bridge sets up `app` quite late meaning it's complicated to rely on it in setup code.

I want to add a method to `Bridge` that constructs and returns a `PrometheusMetrics` instance, having called the setup code to attach it to the `/metrics` path in the `app` instance. Because that `app` instance doesn't exist yet before the run method is called, this can't quite be self-contained. Instead I have to store the metrics instance and run the `addAppServicePath` method later at `run` time.

This logic is further complicated by the fact I don't really know when the caller will be asking for this metrics object. Maybe `run` has already been called so actually I can (and should) register it with `app` now, because I won't get another chance later.

**Comment by leonerd** [comment]
As a possible fix for this I would suggest a method on `Bridge` called something like `getApp` which lazily constructs the app instance, so I can just use that without caring about startup ordering.
